src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11041,12 +11101,22 @@ private LambdaExpressionSyntax ParseLambdaExpression(SyntaxToken asyncToken)
             else
             {
                 var name = this.ParseIdentifierToken();
-                var arrow = this.EatToken(SyntaxKind.EqualsGreaterThanToken);
-                arrow = CheckFeatureAvailability(arrow, MessageID.IDS_FeatureLambda);
-
+                var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
+                SyntaxToken arrow;
+                if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken && this.PeekToken(1).Kind == SyntaxKind.GreaterThanToken)
+                {
+                    var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
+                    arrow = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsGreaterThanToken);
+                    arrow = AddError(arrow, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
```

Comment:
the trick is that you need to consume teh `>` as well (and likely add it as more skipped trivia to the arrow token you pulled out).

Reply:
Got it; I think my params were switched. Pushing